src/components/MapView.js

```diff
@@ -0,0 +1,165 @@
+// src/MapView.js
+import React, {
+  useEffect,
+  useRef,
+  forwardRef,
+  useImperativeHandle,
+  useState,
+  useCallback
+} from 'react';
+import L from 'leaflet';
+import * as topojson from 'topojson-client';
+import 'leaflet/dist/leaflet.css';
+import markerIcon from 'leaflet/dist/images/marker-icon.png';
+import markerShadow from 'leaflet/dist/images/marker-shadow.png';
+
+// Define the default icon for markers
+const defaultIcon = L.icon({
+  iconUrl: typeof markerIcon === 'object' ? markerIcon?.src : markerIcon,
+  shadowUrl:
+    typeof markerShadow === 'object' ? markerShadow?.src : markerShadow,
+  iconSize: [25, 41],
+  iconAnchor: [12, 41],
+  popupAnchor: [1, -34],
+  shadowSize: [41, 41]
+});
+
+const getObjectFromString = (path) =>
```

Comment:
Is it possible to send the topojson directly as parameters so we can remove this check? If it introduces performance issues, let's add error handling when the variable isn't available.

Reply:
Yes, it is possible. Let me fix it.

Reply:
I've added error handling on this commit: [12b8396](https://github.com/akvo/akvo-charts/pull/35/commits/12b8396bd44fd1e4b4c86e2be5841518240f8984)